src/details/ArborX_DetailsUtils.hpp

```diff
@@ -527,8 +527,19 @@ void reallocWithoutInitializing(View &v,
                                 size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG)
 {
   static_assert(View::is_managed, "Can only realloc managed views");
-  v = View(Kokkos::view_alloc(Kokkos::WithoutInitializing, v.label()), n0, n1,
-           n2, n3, n4, n5, n6, n7);
+
+  size_t ranks[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
```

Comment:
I don't like the variable name.  I'd prefer something like `view_extent_orig`.  I am not blocking though.
Similarly I don't like so much `allocation_necessary` because what we check if the view happens to already have the right extents.  In principle, that does not mean that we could not use the memory.  (Yes I know there is no way to  achieve that with the current implementation of Kokkos::View)